integration/integration_test.go

```diff
@@ -0,0 +1,552 @@
+			expectedString: "pod/test-pod created\n",
+		},
+		waitUntilTestPodReady,
+		deleteTestPod,
+	}
+
+	var cmd *command
+	failed := false
+	for _, tt := range commands {
+		switch tt.name {
+		case "Start execsnoop gadget":
+			cmd = tt
+			failed = tt.startCommand(t, failed)
+		case deleteTestPod.name:
+			failed = cmd.stopCommand(t, tt.runCommand(t, failed))
+		default:
+			failed = tt.runCommand(t, failed)
+		}
+	}
+}
+
+func TestMountsnoop(t *testing.T) {
+	commands := []*command{
+		{
+			name:           "Start mountsnoop gadget",
+			cmd:            "$KUBECTL_GADGET mountsnoop -n test-ns",
+			expectedRegexp: `test-pod\s+test-pod\s+mount.*mount\("/mnt", "/mnt", .*\) = -ENOENT`,
+		},
+		{
+			name:           "Run pod which tries to mount a directory",
+			cmd:            "kubectl run --restart=Never --image=busybox -n test-ns test-pod -- sh -c 'while true; do mount /mnt /mnt; done'",
+			expectedRegexp: "pod/test-pod created",
+		},
+		waitUntilTestPodReady,
+		deleteTestPod,
+	}
+
+	var cmd *command
+	failed := false
+	for _, tt := range commands {
+		switch tt.name {
+		case "Start mountsnoop gadget":
+			cmd = tt
+			failed = tt.startCommand(t, failed)
+		case deleteTestPod.name:
+			failed = cmd.stopCommand(t, tt.runCommand(t, failed))
+		default:
+			failed = tt.runCommand(t, failed)
+		}
+	}
+}
+
+func TestNetworkpolicy(t *testing.T) {
+	commands := []*command{
+		{
+			name:           "Run test pod",
+			cmd:            "kubectl run --restart=Never --image=busybox -n test-ns test-pod -- sh -c 'while true; do wget -q -O /dev/null https://kinvolk.io; done'",
+			expectedRegexp: "pod/test-pod created",
+		},
+		waitUntilTestPodReady,
+		{
+			name:           "Run network-policy gadget",
+			cmd:            "$KUBECTL_GADGET network-policy monitor -n test-ns --output ./networktrace.log & sleep 15; kill $!; head networktrace.log",
+			expectedRegexp: `"type":"connect".*"test-ns".*"test-pod"`,
+		},
+		deleteTestPod,
+	}
+
+	failed := false
+	for _, tt := range commands {
+		failed = tt.runCommand(t, failed)
+	}
+}
+
+func TestOpensnoop(t *testing.T) {
+	commands := []*command{
+		{
+			name:           "Start opensnoop gadget",
+			cmd:            "$KUBECTL_GADGET opensnoop -n test-ns",
+			expectedRegexp: `test-ns\s+test-pod\s+test-pod\s+\d+\s+whoami\s+3`,
+		},
+		{
+			name:           "Run pod which calls open()",
+			cmd:            "kubectl run --restart=Never --image=busybox -n test-ns test-pod -- sh -c 'while true; do whoami; done'",
+			expectedRegexp: "pod/test-pod created",
+		},
+		waitUntilTestPodReady,
+		deleteTestPod,
+	}
+
+	var cmd *command
+	failed := false
+	for _, tt := range commands {
+		switch tt.name {
+		case "Start opensnoop gadget":
+			cmd = tt
+			failed = tt.startCommand(t, failed)
+		case cleanupTestNamespace.name:
+			failed = cmd.stopCommand(t, tt.runCommand(t, failed))
+		default:
+			failed = tt.runCommand(t, failed)
+		}
+	}
+}
```

Comment:
I'm wondering if there's a way to avoid having this logic in all the tests. I was thinking about having a general `runCommands` helpers that all tests can use. I see there is some complication because we have two kind of commands: (1) normal: executed with `runCommand`, and (2) start&stop: executed with `startCommand` and `stopCommand`. In order to make this more uniform I think we could only use `runCommand` and add some flags into `command` to indicate how it should behave (run, start or stop). 

It's really difficult to explain by text, so I tried it here -> https://github.com/kinvolk/inspektor-gadget/commit/1a734576f4494a8873bb1a74669ada6d449eb2e8. 

What do you think?